lib/Frontend/FrontendOptions.cpp

```diff
@@ -37,63 +37,74 @@ bool FrontendInputs::shouldTreatAsLLVM() const {
 }
 
 bool FrontendInputs::shouldTreatAsSIL() const {
-  if (hasUniqueInputFilename()) {
+  if (hasUniqueInput()) {
     // If we have exactly one input filename, and its extension is "sil",
     // treat the input as SIL.
     StringRef Input(getFilenameOfFirstInput());
     return llvm::sys::path::extension(Input).endswith(SIL_EXTENSION);
   }
   // If we have one primary input and it's a filename with extension "sil",
   // treat the input as SIL.
-  if (const Optional<StringRef> filename =
-          getOptionalUniquePrimaryInputFilename()) {
-    return llvm::sys::path::extension(filename.getValue())
-        .endswith(SIL_EXTENSION);
+  unsigned silPrimaryCount = numberOfPrimaryInputsEndingWith(SIL_EXTENSION);
+  if (silPrimaryCount == 0)
+    return false;
+  if (silPrimaryCount == primaryInputCount())
```

Comment:
This isn't right; what if there are two input SIL files, both marked as primary?

Reply:
Added assertion about no more than one primary. But I'm wondering if the single input case should be collapsed into a one primary case in general? (Not for this PR.)

Reply:
Mm, the single input might be a single WMO input, which is still considered distinct in a few ways. (It's probably not useful for users, but it's convenient for testing optimizations, since you can optimize much more heavily when you know you're seeing the entire module.)

Reply:
Ah, OK. Tx.